### ISSUE TITLE
Dropout Study

### ISSUE THREAD

**Issue by fmcnallyi3** [fmcnallyi3/icetop-cnn#14]
I would like to see a systematic study on the effect of dropout rates on our reconstruction. A finalized report will test a wide variety of dropout rates and create a simple plot — maybe accuracy or the width of the energy resolution curve vs dropout rate  — in an effort to demonstrate effectiveness.

**Comment by mariahellis** [comment]
Hiii, I’m a physics major involved with the machine learning group from Dr. Frank McNally’s Research in Physics course at Mercer University!

I conducted a dropout study that consisted of 6 different models based on the baseline architecture. After adding dropout parameters, I only changed the dropout percentages, with them being .04, .15, .30, .65, .75, and .90. 

The first set of models trained for 20 epochs, and what I found was that the models with the .04 and .90 parameters (the two extremes) performed the best and were the most comparable, with the other models being significantly less accurate. 

<img width="975" height="341" alt="Image" src="https://github.com/user-attachments/assets/ed8b1289-4bb1-404d-a644-4c9bd9ee3374" />

They also did not train for the full 20 epochs:

<img width="820" height="616" alt="Image" src="https://github.com/user-attachments/assets/cee9a663-cb0b-42ef-b4e7-dcf55c8498b1" />

<img width="959" height="720" alt="Image" src="https://github.com/user-attachments/assets/48238843-9b4d-44b7-8f1c-5aba2ae0ebb1" />

To see if this was an element of randomness, I made 6 more models with the exact same parameters as the original six but ran them for 80 epochs. 

The result was nothing like the 20-epoch result. These 6 new models performed differently; the previous two extremes were now fully comparable / similar in value to the other models. 

<img width="975" height="341" alt="Image" src="https://github.com/user-attachments/assets/7e0080e9-68c8-4824-a97b-d252266092d3" />

Now that the models were performing better, I trained ANOTHER 6 models with the exact same parameters and epochs (80) and got different results that were similar to the 20-epoch test:

<img width="975" height="341" alt="Image" src="https://github.com/user-attachments/assets/e678d050-4c46-486b-9b65-57e6bd470520" />

This means that the first 80-epoch test had an element of randomness being used, which I couldn’t replicate on the second test. To prevent different results using the same inputs, the random seeds for Python, NumPy, TensorFlow should be set early enough to prevent randomness occurring.
